lib/util/Components.js

```diff
@@ -99,9 +99,12 @@ Components.prototype.list = function() {
       component = this.get(node);
     }
     if (component) {
-      usedPropTypes[this._getId(component.node)] = (this._list[i].usedPropTypes || []).filter(function(propType) {
+      var newUsedProps = (this._list[i].usedPropTypes || []).filter(function(propType) {
```

Comment:
Unfortunately it's not so simple in this file. There are many `vars` that cannot be changed to `const`.

Personally I also feel like a file should either use only `var`, or only `let` + `const`. Using a mix of `var`, `let` and `const` is meh.

If you want, I can make some separate PR and transform some of the code-base to ES6? I have used Lebab before (https://github.com/lebab/lebab). I just need to figure out what ES6 features are supported and which aren't. Although we could just start with getting rid of `var` completely?